TWLight/templates/new_base.html

```diff
@@ -10,6 +10,17 @@
     <meta name="viewport" content="width=device-width, initial-scale=1">
     <!-- The above 3 meta tags *must* come first in the head; any other head content must come *after* these tags -->
 
+    <meta property="og:title" content="The Wikipedia Library" />
+    <meta property="og:type" content="website" />
+    <meta property="og:image" content="https://upload.wikimedia.org/wikipedia/commons/6/63/Wikipedia_library_logo.png" />
+    <meta property="og:image:width" content="200" />
+    <meta property="og:image:height" content="200" />
+    <meta property="og:url" content="https://wikipedialibrary.wmflabs.org/" />
+    <meta property="og:description" content="Over 90 of the world's top subscription-only databases, with content in 33 languages, free for Wikipedians of all backgrounds." />
```

Comment:
In `TWLight/templates/login_partial.html`, you can see how we template a source message with the correct arguments here:
```
{% comment %}Translators: Information about The Wikipedia Library.{% endcomment %}
{% blocktranslate trimmed %}
Over 90 of the world's top subscription-only databases,
with content in {{ no_of_languages }} languages, 
free for Wikipedians of all backgrounds
{% endblocktranslate %}
```
For proper i18n, I think you need to:
- use the above message block for `og:description`
- loop through the available languages, and set each one as an `og:locale:alternate` 
- add support for setting the locale via a url parameter (which might be a big task to do correctly; I'd need to do a little research)

Reply:
> loop through the available languages, and set each one as an og:locale:alternate

Hi @jsnshrmn to use this we will need list of `language_TERRITORY` refer: [https://ogp.me/#locale](https://ogp.me/#array:~:text=og%3Alocale%20%2D%20The%20locale%20these%20tags%20are%20marked%20up%20in.%20Of%20the%20format%20language_TERRITORY.%20Default%20is%20en_US.)

we can get a list of languages from the `Language` model in the resources app but it does not have `TERRITORY` do we populate it somewhere in our code or should I try fetching it using some third-party package eg: https://github.com/flyingcircusio/pycountry or is there another way

Reply:
That's a great question. Generally, only the language is required in XPG formatted locale names, eg. `language[_territory[.codeset]][@modifier]` where everything in brackets is optional. I would start with our locales as they are and see if that works.

Because of our integration with translatewiki, we're already crosswalking Django's locales with wikimedia CLDR data, so we could add some machinery to define default territories if necessary, but I wouldn't bother with that complexity unless it's clear that it's required for interoperability.

We do have a few locales with territories defined already that you can use for testing in case they are required:
- en_Gb
- ks-arab
- pt_Br
- sr_Ec
- zh_Hans
- zh_Hant

Reply:
Thank you for the precise explanation @jsnshrmn I have made changes according to your suggestion and pushed a new commit

Reply:
Thank you for the review I have made the changes according to the suggestions using the [django.utils.translation.get_language_from_request()](https://docs.djangoproject.com/en/3.2/ref/utils/#django.utils.translation.get_language_from_request). This has really simplified the task